### DISCUSSION TITLE
smart crop not that smart

### DISCUSSION THREAD

**Discussion by suntong** [libvips/libvips#3767] [Q&A] [answered]
I think I read somewhere that the smart crop is smart enough to detect face and crop around it. Not the case?

Please take a look at the image at
https://raw.githubusercontent.com/h2non/imaginary/master/testdata/smart-crop.jpg

If I use smart crop with `width=500&height=120` (via [imaginary](https://github.com/h2non/imaginary)), then this is what I got and far from what I expected:

![crop2](https://github.com/libvips/libvips/assets/422244/75fe1797-f455-469f-9730-d451b346897d)

I don't know if it is imaginary's problem or libvips's. 
Would someone verify please?

**Comment by jcupitt** [comment, marked as answer]
Hello @suntong,

Yes, the name is maybe a bit misleading! There's no face detector in `smartcrop` -- it searches for edges plus saturated colours plus skintones.

There's a version of smartcrop as a nip2 workspace if you want to see how it works:

![image](https://github.com/libvips/libvips/assets/580843/1e5f81a9-0c65-4537-9eac-5d9d19b7fdd0)

[smartcrop.zip](https://github.com/libvips/libvips/files/13511304/smartcrop.zip)

Load the workspace into nip2 and drag an image into the A1 cell.

**Reply by jcupitt** [reply]
Though now I try that workspace on your image I get a better result:

![image](https://github.com/libvips/libvips/assets/580843/b93d6dc0-76fe-446a-a807-37e4a9fcd6d7)

The parameters have been tuned a few times since the workspace version and I suppose it's no longer working so well for this specific image.

**Reply by suntong** [reply]
> There's no face detector in smartcrop

Oh, I must have mistaken it with something else then. 

Thanks for checking!
